### ISSUE TITLE
I have several questions to consult about tyro's change

### ISSUE THREAD

**Issue by yoker5262** [nerfstudio-project/nerfstudio#2079]
Hello!
nerfstudio/scripts/train.p
My problem is that tyro is not used to receive command line parameters, but to pass parameters in the form of parameters, and as in the command line, only the algorithm name and data path need to be provided to meet the training requirements.
before:
 `def entrypoint():
    """Entrypoint for use with pyproject scripts."""

    tyro.extras.set_accent_color("bright_yellow")
    main(
        tyro.cli(
            AnnotatedBaseConfigUnion,
            description=convert_markup_to_ansi(__doc__),
        )
    )

if __name__ == "__main__":
    entrypoint()`

after:
def entrypoint(name = "in2n",data_path = "data/path"):
    """Entrypoint for use with pyproject scripts."""

    tyro.extras.set_accent_color("bright_yellow")
    main(
        # tyro.cli(
        #     AnnotatedBaseConfigUnion,
        #     description=convert_markup_to_ansi(__doc__),
        # )
        name = name,
        data_path = data_path
        
    )

if __name__ == "__main__":
    entrypoint()

Just an example,I hope you know what I mean,How do I change my code?
Thank you for your guidance.

**Comment by yoker5262** [comment]
I'm Chinese, and that's all translation software can do

**Comment by brentyi** [comment]
Hi @yoker5262!

To clarify my understanding, you want to specify the method type and data directory without going through the command-line, right?

There are a lot of options here, the quickest is to just pass arguments manually into `tyro.cli`, which has an `args=` argument similar to argparse's [`ArgumentParser.parse_args()`](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.parse_args):

```python
main(
    tyro.cli(
        AnnotatedBaseConfigUnion,
        description=convert_markup_to_ansi(__doc__),
        args=[method_name, "--data", data_path]
    )
)
```

You can also grep the code for the `method_configs` dictionary, which is a mapping from string names to config dataclass instances. You can pull out the specific config you want and override the data path from there, and then call the training script's `main()` method on it.

If Chinese is easier feel free to ask/answer questions in Chinese. (at least for configuration stuff)

**Comment by yoker5262** [comment]
非常感谢！您的理解是正确的，并且我在我本地代码中进行了试验，验证无误。

**Comment by yoker5262** [comment]
The question arises again
main(
    tyro.cli(
        AnnotatedBaseConfigUnion,
        description=convert_markup_to_ansi(__doc__),
        args=[method_name, "--data", data_path]
    )
)
This is only for training,How is the ns-render command represented in args。

**Comment by yoker5262** [comment]
The default is train，but i want use ns-render

**Comment by brentyi** [comment]
Maybe refer to `render.py`? That's the script that `ns-render` points to: https://github.com/nerfstudio-project/nerfstudio/blob/main/nerfstudio/scripts/render.py

A similar pattern should work here.

**Comment by yoker5262** [comment]
Is the method of modifying tyro.cli the same? add args?

**Comment by brentyi** [comment]
Yep, that should work. As an alternative, you can also construct the `RenderCameraPath`, `RenderInterpolated`, etc objects manually, and then call the `.main()` method on them.
